lib/ModuleManager.cpp

```diff
@@ -767,8 +768,8 @@ void QoreModuleManager::loadModuleIntern(ExceptionSink& xsink, const char* name,
         else if (QoreDir::folder_exists(modulePath, xsink)) {
             qore_offset_t i = modulePath.rfind(QORE_DIR_SEP);
             // "feature" means pure module name (e.g. "Mime", "CsvUtil" etc.)
-            const char * feature = QoreString(modulePath).extract(++i, &xsink)->c_str();
-            mi = loadSeparatedModule(xsink, modulePath, feature, pgm, reexport, pholder.release(), load_opt & QMLO_REINJECT ? mpgm : 0, load_opt);
+            std::shared_ptr<QoreString> feature(QoreString(modulePath).extract(++i, &xsink));
```

Comment:
there is no point of creating shared_ptr, you are the exclusive owner of the value (you are not passing it anyway where the ownership has to be passed/shared), you can use unique_ptr, it's cheaper to create and it makes more sense.

Reply:
done